src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -16,6 +16,8 @@ namespace Microsoft.OpenApi.Services
     public class OpenApiWalker
     {
         private readonly OpenApiVisitorBase _visitor;
+        private readonly Stack<OpenApiSchema> _schemaLoop = new Stack<OpenApiSchema>();
+        private readonly Stack<OpenApiPathItem> _pathItemLoop = new Stack<OpenApiPathItem>();
```

Comment:
Wow. I didn't realize before today that pathItem can also cause cycles (via Operation -> Callback)!